src/client/Client.js

```diff
@@ -365,6 +366,19 @@ class Client extends BaseClient {
       .then(data => new GuildPreview(this, data));
   }
 
+  /**
+   * Obtains the widget of a guild from Discord, available for all guilds.
+   * @param {GuildResolvable} guild The guild to fetch the widget for
+   * @returns {Promise<Widget>}
+   */
+  async fetchWidget(guild) {
+    const id = this.guilds.resolveID(guild);
+    if (!id) return Promise.reject(new TypeError('INVALID_TYPE', 'guild', 'GuildResolvable'));
```

Comment:
These are in an async method, so you can just throw the errors

Reply:
I don't remember the exact difference, but I know there *is* a difference between the two. Possibly relating to async stack traces?

I think it was @kyranet I discussed it with

Reply:
If something is wrong I can just revert it.
Btw I've seen this throw Error in Client#login (who is async)

Reply:
> I don't remember the exact difference, but I know there is a difference between the two. Possibly relating to async stack traces or being able to .catch the result.

I believe the difference should be unnoticeable in most instances, not sure about performance wise. But, AFAIK, throwing in an async function causes the Promise created by being an async function to reject, whereas `return Promise.reject` resolves the internal async Promise with a rejected Promise, which will still fall through to the catch because that is how Promise chains work.

Reply:
Yeah, youre right, this can be a throw. Its when you return a Promise (eg the .then it was previously) from a non-async function that this is required.